### ISSUE TITLE
FlowTyped Declarations

### ISSUE THREAD

**Issue by colshacol** [AlexGilleran/jsx-control-statements#73]
In usage with Flow, I get errors like this:

<img width="1317" alt="screen shot 2018-01-09 at 12 14 30 am" src="https://user-images.githubusercontent.com/19484365/34707255-1e600ec2-f4d2-11e7-847f-67605bf1bf16.png">

I am wondering that, since the values passed into `condition` are negated to truthy or falsey, shouldn't it accept `any` type?

**Comment by colshacol** [comment]
Current declarations:

<img width="729" alt="screen shot 2018-01-09 at 12 17 29 am" src="https://user-images.githubusercontent.com/19484365/34707333-8a0de75c-f4d2-11e7-852d-d2c9da402f29.png">

What I believe it should be:

<img width="684" alt="screen shot 2018-01-09 at 12 17 42 am" src="https://user-images.githubusercontent.com/19484365/34707332-8a002ba8-f4d2-11e7-9921-c97e48ec0624.png">

**Comment by AlexGilleran** [comment]
Ehhhh I see your point but I actually like that it forces you to put the `!!` in front and say "I definitely want to check for truthiness here, it's not that I've just got the wrong variable/property".

You can just use your own definitions in place of the ones supplied right?

**Comment by colshacol** [comment]
@AlexGilleran Apologies, I am constantly at 200% capacity and forgot to take care of this issue. -- Yeah, I was able to patch it up without any issues.

I think it may boil down to opinion on this, but thinking about what I've used, personally, as well as what I have seen in the community, it is the norm to follow conventions such as

```
if (foo) { ... }
```

or

```
foo && whatever;
```

etc...

In places that values are negated by default, I don't see many explicitly negating.

**Comment by JalilArfaoui** [comment]
Hello @AlexGilleran 

I agree it is a matter of opinion if we should or not have an explicit boolean cast in conditions. 

Except that I think, like @colshacol, that we should have a consistent behavior between `if (foo)` and `<If condition={foo}>`. 

So, since Flow does not raise an error on `if` statement with non-boolean value, I think this link should not either. 

https://flow.org/try/#0PQKgBAAgZgNg9gdzCYAoVUCuA7AxgFwEs5swo44AKADwC4wBnfAJ0OwHMBKep1jsAN6owYQlDA1Og4SLDMApvkzNS1ANwyAvjIVKVYAEQATeVACGmGPkYs27Axu1A

**Comment by draperunner** [comment]
I agree @colshacol and @JalilArfaoui . Since it is so obvious that whatever passed into `<If condition={}>` will be treated as a boolean, like a normal if statement in JavaScript, I think it should be typed as `any`.

In my project we end up casting all statements with `Boolean()` like this: `<If condition={Boolean(...)}>`, which I think is just noisy.
